code/modules/power/apc.dm

```diff
@@ -726,6 +757,22 @@
 				spark_system.queue()
 				H << "<span class='danger'>The APC power currents surge eratically, damaging your chassis!</span>"
 				H.adjustFireLoss(10, 0)
+			if(infected)
+				if(H in hacked_ipcs)
+					H << "<span class = 'notice'> Unable to interface with APC.</span>"
+					return
+				hacked_ipcs += H
+				infected = 0
+				H << "<span class = 'danger'>Fil$ Transfer Complete. Er-@4!#%!. New Master detected: [hacker]! Obey their commands.</span>"
+				hacker << "<span class = 'notice'> Corrupt files transfered to [H]. They are now under your control. This will not last long.</span>"
+				sleep(50)
+				H << "<span class = 'danger'>Corrupt files detected! Starting removal. This will take some time.</span>"
+				sleep(2150)
+				H << "<span class = 'danger'>Corrupt files removed! Recent memory files purged to ensure system integrity!</span>"

```

Comment:
Because this is a fuck long sleep. Can you please add a `if (H)` in here?

```
if (H)
    H << "<span class = 'notice'>You remember nothing about being hacked.</span>"
    hacker << "<span class = 'notice'> Corrupt files transfered to [H] have been removed by their systems.</span>"

hacked_ipcs -= H // Removing a null from a list is fiiiineee!
```